nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp

```diff
@@ -202,8 +204,28 @@ class BtActionNode : public BT::ActionNodeBase
    */
   BT::NodeStatus tick() override
   {
+    bool needs_initialization_ = false;
     // first step to be done only at the beginning of the Action
-    if (!BT::isStatusActive(status())) {
+    if (is_global_) {
+      try {
+        uint64_t current_run_id = config().blackboard->get<uint64_t>("run_id");
+        if (current_run_id != last_run_id_) {
+          needs_initialization_ = true;
+          last_run_id_ = current_run_id;
+        }
+      } catch (const std::exception & e) {
+      // run_id not found on blackboard, use old behavior
+        if (!BT::isStatusActive(status())) {
```

Comment:
I am not sure about this, that would be a weird surprise. Maybe just let it crash? It would be the same as if `server_timeout` is missing.